### PR TITLE
Support local time for scheduled pushes

### PR DESCRIPTION
**Pull request by marvelm**
### Overview
The current behaviour for parsing/formatting push time is inconsistent with [the documentation](http://docs.parseplatform.org/rest/guide/#local-push-scheduling). Parse simply checks that the `push_time` is valid and formats it as an ISO 8601 datetime string. This means that `push_time` is always in UTC even if the creator intended for it to be sent in the local time.

This PR changes the `StatusHandler` so that it strips the trailing "Z" if the timezone isn't specified when the push is created.

### CODE REVIEW

File: src/StatusHandler.js [hunk 1]
```diff
@@ -103,18 +103,28 @@ export function jobStatusHandler(config) {
...
   const setInitial = function(body = {}, where, options = {source: 'rest'}) {
     const now = new Date();
-    let pushTime = new Date();
+
     let status = 'pending';
-    if (body.hasOwnProperty('push_time')) {
+    let formattedPushTime = (new Date()).toISOString();
+    if (body.hasOwnProperty('validatedPushTime')) {
       if (config.hasPushScheduledSupport) {
-        pushTime = body.push_time;
```
Comment (flovilmart) [inline src/StatusHandler.js]: we should keep push_time as the key as well, it's what's expected...

File: src/Controllers/PushController.js [hunk 2]
```diff
@@ -14,10 +14,8 @@ export class PushController {
     }
     // Replace the expiration_time and push_time with a valid Unix epoch milliseconds time
     body.expiration_time = PushController.getExpirationTime(body);
-    const push_time = PushController.getPushTime(body);
-    if (typeof push_time !== 'undefined') {
-      body['push_time'] = push_time;
```
Comment (flovilmart) [inline src/Controllers/PushController.js]: Don't remove that please...

File: src/Controllers/PushController.js [hunk 3]
```diff
@@ -104,21 +105,43 @@ export class PushController {
...
+  }
+
+  static pushTimeHasTimezoneComponent(pushTimeParam) {
+    if (pushTimeParam.indexOf('Z') !== -1) { // 2007-04-05T12:30Z
+      return true;
+    }
+    return pushTimeParam.length === 22; // 2007-04-05T12:30-02:00
+  }
+
+  static formatPushTime({ date, isLocalTime }) {
```
Comment (flovilmart) [inline src/Controllers/PushController.js]: add note that if formats the time, and if passing localTime, strips the 'Z'.

File: spec/PushController.spec.js [hunk 4]
```diff
@@ -979,4 +999,40 @@ describe('PushController', () => {
...
+      expect(PushController.formatPushTime({
+        date: new Date('2017-09-06T17:14:01.048'),
+        isLocalTime: true,
+      })).toBe('2017-09-06T17:14:01.048', 'No timezone');
+      expect(PushController.formatPushTime({
+        date: new Date('2017-09-06'),
+        isLocalTime: true
+      })).toBe('2017-09-06T00:00:00.000', 'YY-MM-DD');
+    });
+  });
```
Comment (flovilmart) [inline spec/PushController.spec.js]: can you please add an edge to edge test, creating a push targeting a single installation with the new format without the TZ? see `PushController.spec.js`

File: resources/npm-git.sh [hunk 5]
```diff
@@ -9,7 +8,8 @@
 # From: https://github.com/graphql/graphql-js/blob/master/resources/npm-git.sh
 
 BUILD_DIR=latest
-
+BRANCH=$(git rev-parse --abbrev-ref HEAD)
```
Comment (flovilmart) [inline resources/npm-git.sh]: can you uncommit that change please? This will break our CI...

Comment (marvelm) [reply resources/npm-git.sh]: Thanks for catching that. 🥇

**Pull request merged**